internal/plugin/includes.go

```diff
@@ -1,18 +1,65 @@
 package plugin
 
 import (
+	"encoding/json"
+	"os"
+	"path/filepath"
 	"strings"
 
 	"go.jetpack.io/devbox/internal/boxcli/usererr"
 	"go.jetpack.io/devbox/internal/nix"
 )
 
-func (m *Manager) parseInclude(include string) (*nix.Package, error) {
+type includable interface {
+	CanonicalName() string
+}
+
+func (m *Manager) parseInclude(include string) (includable, error) {
 	includeType, name, _ := strings.Cut(include, ":")
-	if includeType != "plugin" {
-		return nil, usererr.New("unknown include type %q", includeType)
-	} else if name == "" {
+	if name == "" {
 		return nil, usererr.New("include name is required")
+	} else if includeType == "plugin" {
+		return nix.PackageFromString(name, m.lockfile), nil
+	} else if includeType == "path" {
+		absPath := filepath.Join(m.ProjectDir(), name)
+		return newLocalPlugin(absPath)
+	}
+	return nil, usererr.New("unknown include type %q", includeType)
+}
+
+type localPlugin struct {
+	name string
+	path string
+}
+
+func newLocalPlugin(path string) (*localPlugin, error) {
+	content, err := os.ReadFile(path)
+	if err != nil {
+		return nil, err
+	}
+	m := map[string]any{}
+	if err := json.Unmarshal(content, &m); err != nil {
+		return nil, err
 	}
-	return nix.PackageFromString(name, m.lockfile), nil
+	name, ok := m["name"].(string)
+	if !ok || name == "" {
+		return nil,
+			usererr.New("plugin %s is missing a required field 'name'", path)
+	}
```

Comment:
I'd suggest setting some constraints on allowable names. Maybe:
1. specify that they must belong to `[a-z][A-Z]-` 
2. length limit of 32 or 64 characters
?

I'd worry that having no constraints would get "creatively" used

Reply:
done